src/main/include/wrapperclasses/Motor.h

```diff
@@ -0,0 +1,57 @@
+#pragma once 
+
+class Motor {
+    public:
+        Motor(); 
+
+        virtual void SetPercent(double percent) = 0;
+        virtual void StopMotor() = 0; 
+
+        virtual void Configure() = 0; 
+
+        // ENCODER FUNCTIONS // 
+
+        virtual double GetRelativeVelocity() = 0; 
+        virtual double GetRelativePosition() = 0; 
+
+        virtual double SetRelativePosition(double pos) = 0; 
+
+        // END ENCODER FUNCTIONS // 
+
+        // CONFIGURE SETTINGS // 
+
+        virtual void configure() = 0; 
+        virtual void setPID(double p, double i, double d, double ff) = 0; 
+        virtual void setPID(double p, double i, double d) = 0; 
+
+        virtual void setForwardSoftLimit(double limit) = 0; 
+        virtual void setReverseSoftLimit(double limit) = 0; 
+        virtual void enableForwardSoftLimit(bool enab) = 0; 
+        virtual void enableReverseSoftLimit(bool enab) = 0; 
+
+        virtual void setInverted(bool b) = 0; 
+
+        // struct PIDContainer {
+        //     double P; 
+        //     double I; 
+        //     double D; 
+        //     double ff; 
+        // };
+
+        //PIDContainer PID; 
+
+        struct ConfigParamDataType {
```

Comment:
Other config parameters we probably care about:
- For base config:
  - Idle mode
  - Current limit
  - Voltage compensation
- For absolute encoder:
  - Inverted
  - Position conversion factor
  - Velocity conversion factor
- For PID:
  - Feedback Sensor
  - Min output and max output (or output range)
  - Maybe input wrapping
- For relative encoder:
  - Counts per revolution
  - Inverted
  - Position conversion factor
  - Velocity conversion factor
- Limit switches
  - Forward/reverse limit switch enabled
  - Forward/reverse limit switch type
  -